tutorials/epochs/15_baseline_regression.py

```diff
@@ -0,0 +1,286 @@
+mne.viz.plot_compare_evokeds(dict(auditory=reg_aud, visual=reg_vis),
+                             title="Regression-based", **kwargs)
+
+# %%
+# They look pretty similar, but there are some subtle differences in how far
+# apart the two conditions are (e.g., around 400-500 ms).
+#
+# Plot the scalp topographies and difference waves
+# ================================================
+#
+# Now let's compare the
+# scalp topographies for the traditional and regression-based approach. We'll
+# do this by computing the difference between conditions:
+
+diff_traditional = mne.combine_evoked([trad_aud, trad_vis], weights=[1, -1])
+diff_regression = mne.combine_evoked([reg_aud, reg_vis], weights=[1, -1])
+
+# %%
+# Before we plot, let's make sure we get the same color scale for both figures:
+
+vmin = min(diff_traditional.get_data().min(),
+           diff_regression.get_data().min()) * 1e6
+vmax = max(diff_traditional.get_data().max(),
+           diff_regression.get_data().max()) * 1e6
+topo_kwargs = dict(vmin=vmin, vmax=vmax, ch_type='eeg',
+                   times=np.linspace(0.05, 0.45, 9))
+
+diff_traditional.plot_topomap(title="Traditional", **topo_kwargs)
+
+# %%
+diff_regression.plot_topomap(title="Regression-based", **topo_kwargs)
+
+# %%
+# We can see that the regression-based approach shows *stronger* difference
+# between conditions early on (around 100-150 ms) and *weaker* differences
+# later (around 250-350 ms, and again around 450 ms). This is also reflected in
+# the difference waves themselves: notice how the regression-based difference
+# wave is *further from zero* around 150 ms but *closer to zero* around 250-350
+# ms.
+
+title = "Difference in evoked potential (auditory minus visual)"
+fig = mne.viz.plot_compare_evokeds(dict(Traditional=diff_traditional,
+                                        Regression=diff_regression),
+                                   title=title, **kwargs)
+
+# %%
+# Examine the interaction term
+# ============================
+#
+# Finally, let's look at the interaction term from the regression model. This
+# tells us whether the effect of the baseline period is different in the visual
+# trials versus its effect in the auditory trials. Here we'll add a horizontal
+# line at zero, indicating the assumption that there ought to be no difference
+# (i.e., baselines should not be systematically higher in one type of trial,
+# and there should not be a difference in how long the effect of the baseline
+# persists through time in each type of trial).
+
+interaction_effect = reg_model['baseline:visual'].beta
+interaction_effect.plot(picks=ch, hline=[0.], units=dict(eeg=r'$\beta$ value'),
+                        titles=dict(eeg=ch), selectable=False)
+
+
+# %%
+# Indeed, the interaction beta weights are rather small and seem to fluctuate
+# randomly around zero, suggesting that there is no systematic difference in
+# the effect of the baseline on our two trial types.
```

Comment:
Yep, exactly! Regression-based baseline shouldn't find a difference between conditions in many traditional experiments, which are designed to avoid such confounds. One of the big advantages is that you can still do experiments where such confounds are unavoidable (e.g. more naturalistic experiments).